openlibrary/plugins/openlibrary/js/lists/index.js

```diff
@@ -438,3 +447,113 @@ function updateAlreadyList(listKey, listTitle, coverUrl, seedKey) {
 
     return li;
 }
+
+/**
+ * Initializes asynchronous list widget loading.
+ *
+ * Get references to all loading indicators and begins their animations.
+ * Gets the edition or work key for the book referenced by the dropper, and
+ * fetches HTML partials for the key.
+ *
+ * Once the partials are received, the loading indicators are replaced with the
+ * partials, and click listeners are added to the new elements
+ * @param {HTMLElement} dropperList Container for dropper list items
+ * @param {HTMLElement} activeList Container for active lists list items
+ */
+export function initListLoading(dropperList, activeList) {
+    const loadingIndicators = dropperList ? [dropperList.querySelector('.loading-ellipsis')] : []
+    if (activeList) {
+        loadingIndicators.push(activeList.querySelector('.loading-ellipsis'))
+    }
+    const intervalId = initLoadingAnimation(loadingIndicators)
+
+    let key
+    if (dropperList) {  // Not defined for logged out patrons
+        if (dropperList.dataset.editionKey) {
+            key = dropperList.dataset.editionKey
+        } else if (dropperList.dataset.workKey) {
+            key = dropperList.dataset.workKey
+        }
+    }
+
+    if (key) {
+        fetchPartials(key, function(data) {
+            clearInterval(intervalId)
+            replaceLoadingIndicators(dropperList, activeList, data)
+        })
+    } else {
+        removeChildren(dropperList, activeList)
+    }
+}
+
+/**
+ * Animates ellipsis that follows the word "Loading"
+ *
+ * A new dot is appended every 1.5 seconds, until there
+ * is a full ellipsis.  This cycle repeats indefinately.
+ *
+ * Returns an interval ID, which should be used to terminate
+ * the `setInterval` call.
+ * @param {HTMLElement[]} loadingIndicators References to the loading indicators
+ * @returns {number} Interval ID returned by the internal `setInterval` call
+ */
+function initLoadingAnimation(loadingIndicators) {
+    let count = 0;
+    const intervalId = setInterval(function() {
+        let ellipsis = ''
+        for (let i = 0; i < count % 4; ++i) {
+            ellipsis += '.'
+        }
+        for (const elem of loadingIndicators) {
+            elem.innerText = ellipsis
+        }
```

Comment:
Oh if we update all of them at the same time, does that mean they all have to finish at exactly the same time?